backend-shared/db/app_project_managed_env_test.go

```diff
@@ -66,4 +76,86 @@ var _ = Describe("AppProjectManagedEnvironment Test", func() {
 
 	})
 
+	Context("Test CountAppProjectManagedEnvironmentByClusterUserID function", func() {
+		It("should return 0 if there are no clusterusers", func() {
+			appProjEnv := &db.AppProjectManagedEnvironment{}
+			rows, err := dbq.CountAppProjectManagedEnvironmentByClusterUserID(ctx, appProjEnv)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(rows).To(BeZero())
+		})
+
+		It("should return the number of rows of AppProjectManagedEnvironment for a given clusteruser", func() {
+			By("add sample AppProjectManagedEnvironment with a given clusteruser")
+			expectedRowCount := 1
+			_, managedEnvironment, _, _, _, err := db.CreateSampleData(dbq)
+			Expect(err).ToNot(HaveOccurred())
+
+			var clusterUser = &db.ClusterUser{
+				Clusteruser_id: "test-user-application",
+				User_name:      "test-user-application",
+			}
+			err = dbq.CreateClusterUser(ctx, clusterUser)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Verify whether AppProjectManagedEnvironment is created")
+			appProjectManagedEnv := db.AppProjectManagedEnvironment{
+				Clusteruser_id:         clusterUser.Clusteruser_id,
+				Managed_environment_id: managedEnvironment.Managedenvironment_id,
+				SeqID:                  int64(seq),
+			}
+
+			err = dbq.CreateAppProjectManagedEnvironment(ctx, &appProjectManagedEnv)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("verify if the row count matches with the expected AppProjectManagedEnvironment count")
+			rowCount, err := dbq.CountAppProjectManagedEnvironmentByClusterUserID(ctx, &appProjectManagedEnv)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(rowCount).To(Equal(expectedRowCount))
+		})
+	})
+
+	Context("Test ListAppProjectManagedEnvironmentByClusterUserId function", func() {
+		It("should return AppProjectManagedEnvironment with the given clusterUserID", func() {
+			_, managedEnvironment, _, _, _, err := db.CreateSampleData(dbq)
+			Expect(err).ToNot(HaveOccurred())
+
+			var clusterUser = &db.ClusterUser{
+				Clusteruser_id: "test-user-application",
+				User_name:      "test-user-application",
+			}
+			err = dbq.CreateClusterUser(ctx, clusterUser)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Verify whether AppProjectManagedEnvironment is created")
+			appProjectManagedEnv := db.AppProjectManagedEnvironment{
+				Clusteruser_id:         clusterUser.Clusteruser_id,
+				Managed_environment_id: managedEnvironment.Managedenvironment_id,
+				SeqID:                  int64(seq),
+			}
+
+			err = dbq.CreateAppProjectManagedEnvironment(ctx, &appProjectManagedEnv)
+			Expect(err).ToNot(HaveOccurred())
+
+			appProjectManagedEnvs := []db.AppProjectManagedEnvironment{}
```

Comment:
Use this instead?  `var appProjectManagedEnvs []db.AppProjectManagedEnvironment`